src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1403,10 +1403,35 @@ void LogCommand()
                 {
                     sb.Append('\t').Append("Parameters:");
                     for (var i = 0; i < p.Count; i++)
-                        sb.Append("\t$").Append(i + 1).Append(": ").Append(Convert.ToString(p[i].Value, CultureInfo.InvariantCulture));
+                    {
+                        if (p[i].Value.GetType().IsArray)
```

Comment:
Just cast value to a `IList` and be happy. All single dimension arrays implement this interface, so you will able to iterate over them. While we support multidimensional, they are rare, so it's up to you to think implement it or not. If you're interested, take a look at the array handler and how it iterates over multidimensional arrays.

Reply:
Good point, I'll sort this out and push the changes asap.

Reply:
```csharp
if (p[i].Value is IList list)
    // wrtie it like it's an array
else
    // a scalar if we don't care about multidimensional array
```

By the way, `Value` could be `null` in some cases.

Reply:
Thanks @YohDeadfall, apologies the changes took so long to do but I got caught up in some other urgent matters and haven't had too much spare time on my hands. I also ran out of time to look into the multidimensional arrays but will have a look when i get a chance and possibly do a separate PR at a later date. Thanks for all the pointers, appreciate it.